File: clases 25-11-2025/Tipos de clases c#/clase normal con interfaz/Profesor.cs
public class Profesor : IProfesor
{
    public string Nombre { get; set; }
    public string Materia { get; set; }

    public Profesor(string nombre, string materia)
    {
        Nombre = nombre;
        Materia = materia;
    }

    public void MostrarProfesor()
    {
        Console.WriteLine($"Profesor: {Nombre}, Materia: {Materia}");
    }
}

File: clases 25-11-2025/Tipos de clases GO/clase normal con interfaz/Profesor.java
public class Profesor implements IProfesor {
    public String nombre;
    public String materia;

    // Constructor con parámetros
    public Profesor(String nombre, String materia) {
        this.nombre = nombre;
        this.materia = materia;
    }

    @Override
    public void mostrarProfesor() {
        System.out.println("Profesor: " + nombre + ", Materia: " + materia);
    }
}
